src/client/main.cpp

```diff
@@ -27,7 +58,41 @@ int main(int argc, char* argv[])
         client.connectAndListen(config.network.server_ip);
     }
 
-    client.start(context);
+
+    if (client.init() == -1) {
+        exit(EXIT_FAILURE);
+    }
+    
+    GLFWwindow* window = client.getWindow();
+    if (!window) exit(EXIT_FAILURE);
+
+    // Setup callbacks.
+    set_callbacks(window);
+    // Setup OpenGL settings.
+    set_opengl_settings();
+
+    // Initialize the shader program; exit if initialization fails.
+    // if (!Window::initializeProgram()) exit(EXIT_FAILURE);
+    // Initialize objects/pointers for rendering; exit if initialization fails.
+    // if (!Window::initializeObjects(argc, argv)) exit(EXIT_FAILURE);
```

Comment:
Are these initializeProgram functions in our code? If not, I think we can probably remove this block.

Reply:
No, those are just placeholders for init functions, which should already take care of everything we need for now. I'll clean up some of the commented code